cmd/lakectl/cmd/ingest.go

```diff
@@ -21,51 +49,77 @@ var ingestCmd = &cobra.Command{
 		dryRun := MustBool(cmd.Flags().GetBool("dry-run"))
 		from := MustString(cmd.Flags().GetString("from"))
 		to := MustString(cmd.Flags().GetString("to"))
+		concurrency := MustInt(cmd.Flags().GetInt("concurrency"))
 		lakefsURI := MustParsePathURI("to", to)
 
+		// initialize worker pool
+		var wg sync.WaitGroup
+		wg.Add(concurrency)
+		requests := make(chan *stageRequest)
+		responses := make(chan *stageResponse)
+		for w := 0; w < concurrency; w++ {
+			go stageWorker(&wg, requests, responses)
+		}
+
 		summary := struct {
 			Objects int64
 			Bytes   int64
 		}{}
-		client := getClient()
-		err := store.Walk(ctx, from, func(e store.ObjectStoreEntry) error {
-			if dryRun {
-				Fmt("%s\n", e)
+
+		// iterate entries and feed our pool
+		go func() {
+			err := store.Walk(ctx, from, func(e store.ObjectStoreEntry) error {
+				if dryRun {
+					Fmt("%s\n", e)
+					return nil
+				}
+				key := e.RelativeKey
+				if lakefsURI.Path != nil && *lakefsURI.Path != "" {
+					path := *lakefsURI.Path
+					if strings.HasSuffix(*lakefsURI.Path, "/") {
```

Comment:
This can be done once outside the loop - just set `prefix` with the right value

Reply:
Done